src/main/java/seedu/address/model/person/MedCon.java

```diff
@@ -12,18 +12,18 @@
 public class MedCon implements Comparable<MedCon> {
 
     public static final String VALIDATION_REGEX = "\\p{Alnum}+(\\s\\p{Alnum}+)*";
-    public final String value;
+    public final String medConName;
 
     /**
      * Constructs a {@code MedCon} object with the specified medical condition string.
      *
-     * @param medCon The medical condition of the patient
+     * @param medConName The medical condition of the patient
      */
-    public MedCon(String medCon) {
-        requireNonNull(medCon);
-        checkArgument(!medCon.isEmpty(), MESSAGE_EMPTY_FIELD);
-        checkArgument(isValidMedCon(medCon), MESSAGE_CONSTRAINTS_LENGTH);
-        value = medCon;
+    public MedCon(String medConName) {
+        requireNonNull(medConName);
+        checkArgument(!medConName.isEmpty(), MESSAGE_EMPTY_FIELD);
+        checkArgument(isValidMedConName(medConName), MESSAGE_CONSTRAINTS_ALPHANUMERIC_LENGTH);
```

Comment:
Good use of multiple error messages to provide users more info

Reply:
But might be violating `DRY` since `isValidMedConName` is already checking nullity. Why not let `isValidMedConName` check and throw the Exceptions (ie push the `checkArgument` calls into `isValidMedConName`), and let the Exception propagate up and out of the constructor via `isValidMedConName` instead. It will reduce coverage misses too.

(if you choose this route, rename the function)

(take a look at `Appointment::checkTimePeriodValid`)

Reply:
If you wanna quickly make an _issue_ for the PR, just press the 3 dots and `Reference in new issue`